CHANGELOG.md

```diff
@@ -24,6 +24,8 @@ All notable changes for each version of this project will be documented in this
     - Removed `leftButtonColor`, `leftButtonBackgroundColor` `rightButtonColor`, and `rightButtonBackgroundColor`  properties.
 - `IgxGrid`, `IgxTreeGrid`, `IgxHierarchicalGrid`
     - Enhanced the advanced filtering to emit the `filtering` event when filters are applied.
+- `IgxCombo`, `IgxSimpleCombo`:
+    - Introduced support for hiding the clear icon button when the custom clear icon template is empty.
```

Comment:
That's an actual fix though :) Not sure it needs a changelog entry

Reply:
Could you confirm whether this adjustment should be considered a feature or a fix for versions 16.1, 17.1, and master?

Regarding the approach, I'd like to understand your preference: should I prioritize this solution proposed in the PR, or do you recommend implementing the suggested alternative?

Reply:
Hm, might've confused that with the somewhat similar Data Picker toggle/clear icons, because I remember discussing with the team that the default could be removed by providing an empty template for those for sure. Sort of assumed it's the same for the Combo, but looks to be a bit older and doesn't seem to be documented either, so it might as well be new feature :)

As for the implementation: If the main issue is the remaining padding that needs to be removed, I tested this just briefly my suggestion in dev tools on a live sample but it should do the trick. The current version that creates an embedded view on the fly just to check its contents and discards it, while fun, could be a non-trivial amount of work for such an issue depending on the template. Also consider that being a getter will run on every change detection cycle, so would realistically leave that a last resort if nothing else works. I think we even have precedent with taking a ref and checking childNodes in the toolbar that can do the same check, but if a simple style would do too - I think that's our best option.